### ISSUE TITLE
[BUG 1.16.5] Digger zombie cannot dig hand breakable blocks that require no tool when "proper tool only" set to true

### ISSUE THREAD

**Issue by NighthowlerStudios** [Insane96/EnhancedAI#85]
**Describe the bug**  
"Digger With Proper Tool Only" does not work as intended.  

I have set the boolean to true, and the config comment states "blocks that require no tool will still be broken".  the zombie will not break a wooden fence, though it was allowed to before when that boolean was set to false.

to ensure every zombie spawns with the ability to break blocks, I set "digger chance" to 1.0

I'm not sure if this bug is because wooden fences are hand breakable even though they have tools to break them faster.  I have not seen the code for that before.

Additionally note that "Digger Tool Only" is kept to false, which should theoretically allow them to use hands, but it still won't for some reason.  Setting this to true prevents the issue as they wouldn't be allowed to attempt in the first place.

**Versions (don't use "latest"):**  
 - Minecraft: 1.16.5
 - Forge: 36.2.34
 - Enhanced AI: 1.2.3
 - InsaneLib: 1.4.2

**Steps to reproduce**  
1.   Set digger chance to 1.0 to ensure all zombies spawn with the ability
2.   Set "digger with proper tool only" to true
3.   Place a zombie in a fence and a villager in a nearby fence
4.   The zombie will not be able to break the fence
5.   Shut down the game, set "digger with proper tool only" to false
6.   Restart and the zombie will be able to break the fence as expected.

**Can the issue be reproduced with EnhancedAI only?**  
Yes. 
![image](https://user-images.githubusercontent.com/82226641/227801056-59341736-bedd-486d-954d-a941a1915bff.png)

**Comment by NighthowlerStudios** [comment]
New information:

for whatever reason, the same bug occurs if the boolean is set to true, even if the zombie is holding the correct tool.  I used a command block to summon one manually.  Holding a diamond pickaxe, it refused to break the stone wall in front of it blocking its path to a villager

**Comment by Insane96** [comment]
The tool must be in the off-hand for "Tool Only" to work.

Gonna test this in 1.19.3 as the code hasn't changed.

**Comment by NighthowlerStudios** [comment]
> The tool must be in the off-hand for "Tool Only" to work.

understood.  "tool only" does seem to work. 
 but the "proper tool" behaviour still seems broken, right?  i would assume an empty hand, even if it is the offhand, against dirt would have worked.

**Comment by Insane96** [comment]
> but the "proper tool" behaviour still seems broken, right? i would assume an empty hand, even if it is the offhand, against dirt would have worked

As you report it, yes, seems to be broken as an empty hand should work.

**Comment by NighthowlerStudios** [comment]
I have found that indeed the issue seemed to actually be caused by SpecialAI instead of Enhanced AI.  I was able to prevent the crash by making sure its ability blacklist was applied to all entities.

However the empty hand issue is still very much prevalent.  I am unsure if you wish for me to open a separate issue and let that be tracked there and close this one, or to just keep this one open.

Sorry it took me so long to get back to you, struggling with university workloads at the moment ^^'
